src/lib/core/redux/slices/rtcConnection/actions.ts

```diff
@@ -10,3 +14,46 @@ export const rtcEvents = {
     rtcManagerDestroyed: createRtcEventAction<void>("rtcManagerDestroyed"),
     streamAdded: createRtcEventAction<RtcStreamAddedPayload>("streamAdded"),
 };
+
+type RtcAnalyticsCustomEvent = {
+    actionType: string;
+    rtcEventName: string;
+    getValue: (state: RootState) => unknown;
+    getOutput: (value: unknown) => unknown;
+};
+
+export const rtcAnalyticsCustomEvents: { [key: string]: RtcAnalyticsCustomEvent } = {
+    audioEnabled: {
+        actionType: "localParticipant/doEnableAudio/fulfilled",
+        rtcEventName: "audioEnabled",
+        getValue: (state: RootState) => selectIsMicrophoneEnabled(state),
+        getOutput: (value) => ({ enabled: value }),
```

Comment:
Let's import the action creators and use them directly.

```suggestion
        action: doEnableAudio.fulfilled,
        rtcEventName: "audioEnabled",
        getValue: (state: RootState) => selectIsMicrophoneEnabled(state),
        getOutput: (value) => ({ enabled: value }),
```